pkg/canary/daemonset_controller.go

```diff
@@ -74,17 +74,15 @@ func (c *DaemonSetController) ScaleFromZero(cd *flaggerv1.Canary) error {
 
 // Initialize creates the primary DaemonSet, scales down the canary DaemonSet,
 // and returns the pod selector label and container ports
-func (c *DaemonSetController) Initialize(cd *flaggerv1.Canary, skipLivenessChecks bool) (err error) {
+func (c *DaemonSetController) Initialize(cd *flaggerv1.Canary) (err error) {
 	err = c.createPrimaryDaemonSet(cd)
 	if err != nil {
 		return fmt.Errorf("createPrimaryDaemonSet failed: %w", err)
 	}
 
 	if cd.Status.Phase == "" || cd.Status.Phase == flaggerv1.CanaryPhaseInitializing {
-		if !skipLivenessChecks && !cd.SkipAnalysis() {
```

Comment:
Removing the SkipAnalysis condition changes the current behaviour. Is this intentional? if so please add a separate commit with it.

Reply:
I don't think this changes the behavior since `skipLivenessChecks` is always `False` and we have always entered this block

but it's better to have a separate commit as you said

Reply:
I misunderstood this condition - this is not intentional so I am going to revert this change

Reply:
force pushed the reverted version